### PR TITLE
[Optimus] Respect includePath param

### PR DESCRIPTION
**Pull request by Kyle-Ye**
Bug/issue #, if applicable: 

## Summary

Respect includePath param

Since  we call  `range.diagnosticDescription(includePath: false)` here, we should respect the false here instead of ignore it.

```swift
    private mutating func dump(_ markup: Markup, customDescription: String? = nil) {
        indent(markup)
        result += "\(type(of: markup))"
        if options.contains(.printSourceLocations),
            let range = markup.range {
            result += " @\(range.diagnosticDescription(includePath: false))"
        }
        if options.contains(.printUniqueIdentifiers) {
            if markup.parent == nil {
                result += " Root #\(markup._data.id.rootId)"
            }
            result += " #\(markup._data.id.childId)"
        }
        if let customDescription = customDescription {
            if !customDescription.starts(with: "\n") {
                result += " "
            }
            result += "\(customDescription)"
        }
        increasingDepth(markup)
    }
```

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Ran the `./bin/test` script and it succeeded

### CODE REVIEW

File: Sources/Markdown/Infrastructure/SourceLocation.swift [hunk 1]
```diff
@@ -90,14 +90,13 @@ extension SourceRange {
...
-    public func diagnosticDescription(includePath: Bool = true) -> String {
-        let path = lowerBound.source.map {
-            $0.path.isEmpty
-                ? ""
-                : "\($0.path):"
-        } ?? ""
-
-        var result =  "\(path)\(lowerBound)"
+    func diagnosticDescription(includePath: Bool = true) -> String {
+        let path = includePath ? lowerBound.source?.path ?? "" : ""
```
Comment (Kyle-Ye) [inline Sources/Markdown/Infrastructure/SourceLocation.swift:94]: Core of this PR

**Review by QuietMisdreavus — APPROVED** [review]
This looks great, thanks!

**Comment by QuietMisdreavus** [comment]
@swift-ci Please test

**Comment by QuietMisdreavus** [comment]
@swift-ci Please test

**Pull request merged**